packages/bitcoin-node-wallet-provider/lib/BitcoinNodeWalletProvider.js

```diff
@@ -0,0 +1,122 @@
+import WalletProvider from '@liquality/wallet-provider'
+import JsonRpcProvider from '@liquality/jsonrpc-provider'
+import BitcoinNetworks from '@liquality/bitcoin-networks'
+import * as bitcoin from 'bitcoinjs-lib'
+import { sha256 } from '@liquality/crypto'
+import { Address, addressToString } from '@liquality/utils'
+
+import { version } from '../package.json'
+
+const BIP70_CHAIN_TO_NETWORK = {
+  'main': BitcoinNetworks.bitcoin,
+  'test': BitcoinNetworks.bitcoin_testnet,
+  'regtest': BitcoinNetworks.bitcoin_regtest
+}
+
+const ADDRESS_TYPES = ['legacy', 'p2sh-segwit', 'bech32']
+
+export default class BitcoinNodeWalletProvider extends WalletProvider {
+  constructor (network, uri, username, password, addressType = 'bech32') {
+    super()
+    if (!ADDRESS_TYPES.includes(addressType)) {
+      throw new Error(`addressType must be one of ${ADDRESS_TYPES.join(',')}`)
+    }
+    this.addressType = addressType
+    this.network = network
+    if (this.network.name === BitcoinNetworks.bitcoin.name) {
```

Comment:
Same with this logic. probably we can accommodate something like this in networks file. Just to reduce code duplication.

Reply:
Bare with me on this one, actually our own custom `BitcoinNetworks` objects will be removed and the providers will use the BitcoinJsLib provided version. At which point, this mapping won't be required. Probably `BitcoinNetworks` will just alias to `bitcoinjs-lib.networks`

Just trying to get the main segwit part through, there are many improvements like this i am still pending to make

Reply:
Our custom network object now just aliases onto the BitcoinJsLib object with a few additional properties. 
Due to this, we no longer require 2 different objects to deal with this and this mapping is gone.